prow/test/integration/internal/fakepubsub/fakepubsub.go

```diff
@@ -0,0 +1,94 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package fakepubsub
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+
+	"cloud.google.com/go/pubsub"
+	"github.com/sirupsen/logrus"
+	"google.golang.org/api/option"
+	"k8s.io/test-infra/prow/pubsub/subscriber"
+)
+
+type PubSubMessageForSub struct {
+	Attributes map[string]string
+	Data       subscriber.ProwJobEvent
+}
+
+type Client struct {
+	projectID    string
+	pubsubClient *pubsub.Client
+}
+
+func NewClient(projectID, pubsubEmulatorHost string) (*Client, error) {
+	// Force this client to always talk to the fakepubsub service running in the
+	// test KIND cluster.
+	os.Setenv("PUBSUB_EMULATOR_HOST", pubsubEmulatorHost)
```

Comment:
would like to point out that this sets up a global env throughout the entire test process, which might or might not cause side effect. It would be ideal if this could be an option to `pubsub.NewClient`, but reading through https://cloud.google.com/pubsub/docs/emulator and there is not clear indicator of such an option.

One thing not super clear from the emulator doc, is that at which step this env var is needed, if it's not required for creating the client, aka the emulator endpoint value is assigned to a field in the client, then it might be safe to unset the env var right after the `pubsub.NewClient` call

Reply:
I don't see anything either on https://pkg.go.dev/cloud.google.com/go/pubsub#hdr-Emulator. From reading those docs I think you have to have the environment variable set at all times.

I think it's fine for now to have all pubsub client calls go to the fakepubsub service for all of our integration tests though. If we do need to talk to the real Pub/Sub service, we can revisit this.

Reply:
Oops, I do see a field being set actually in https://github.com/googleapis/google-cloud-go/blob/e43c095c94e44a95c618861f9da8f2469b53be16/pubsub/pubsub.go#L126. Will update this.

Reply:
nice

Reply:
Done